src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -64,15 +81,42 @@ private static void initHttpClient() {
         new SSLConnectionSocketFactory(
             sslContext, new String[] {"TLSv1.2"}, null, new DefaultHostnameVerifier());
 
+    // Set connectionTimeout which is the timeout until a connection with the server is established
+    // Set connectionRequestTimeout which is the time to wait for getting a connection from the
+    // connection pool
+    // Set socketTimeout which is the max time gap between two consecutive data packets
+    RequestConfig requestConfig =
+        RequestConfig.custom()
+            .setConnectTimeout(
+                (int)
+                    TimeUnit.MILLISECONDS.convert(
+                        DEFAULT_CONNECTION_TIMEOUT_MINUTES, TimeUnit.MINUTES))
```

Comment:
technically connect timeout is unrelated to the issue at hand. what is the default value for this?

Reply:
I got this commit from an already merged commit to master #104 

default values are -1, which means undefined, (From the documentation it is not clear what exactly is undefined, https://hc.apache.org/httpcomponents-client-4.5.x/current/httpclient/apidocs/org/apache/http/client/config/RequestConfig.html#getConnectTimeout())